### PR TITLE
Add codecheck rule for image files

### PR DESCRIPTION
**Pull request by gunchleoc**
This new rule checks whether image files loaded via `g_gr->images().get()` exist. It can only handle string literals though.

### CODE REVIEW

**Comment by gunchleoc** [comment]
I managed to hack the base path for Travis, so this is ready for review now.

File: cmake/codecheck/rules/images_exist.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+"""
+
+import os.path
+import re
+
+
+FIND_LITERALS = re.compile(r"""g_image_cache->get\("(.*?)"\)""")
+# We skip pointers and function calls here
+FIND_VARIABLE_NAMES = re.compile(
+    r"""g_image_cache->get\(([0-9a-zA-Z_-]*?)\)""")
```
Comment (Niektory) [inline cmake/codecheck/rules/images_exist.py]:
```suggestion
    r"""g_image_cache->get\(([0-9a-zA-Z_]+?)\)""")
```
Just a small tweak.

File: cmake/codecheck/rules/images_exist.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+                    variables.append(match)
+
+    # Try to identify images that are fetched by variable name.
+    # We won't be able to check everything, e.g. when the filename is passed via
+    # a function parameter
+    for variable in variables:
+        found = False
+        no_matches = 0
+        found_on_line = 1
+        FIND_VARIABLES = re.compile(variable + r"\w+=\w+\"(.*?)\"")
```
Comment (Niektory) [inline cmake/codecheck/rules/images_exist.py]:
```suggestion
        FIND_VARIABLES = re.compile(r"\b" + variable + r"""\s*=\s*"(.+?)";""")
```
This was not matching anything so I changed it.

**Review by Niektory — CHANGES_REQUESTED** [review]

File: cmake/codecheck/rules/images_exist.py (hunk 1 above)
Comment (gunchleoc) [reply cmake/codecheck/rules/images_exist.py]: Any particular reason you are removing the `-` character here? Changing `*` to `+` looks good.

Comment (Niektory) [reply cmake/codecheck/rules/images_exist.py]: Variable names can't have `-`, so I don't know what it's doing there.

Comment (gunchleoc) [reply cmake/codecheck/rules/images_exist.py]: Oops, let's get rid of it then.

**Review by Niektory — APPROVED** [review]

**Pull request merged**